rom/dev/src/main.rs

```diff
@@ -126,15 +128,22 @@ pub extern "C" fn rom_entry() -> ! {
 
             /// Ecc384 Engine
             ecc384: &mut env.ecc384,
+
+            /// SHA Acc lock state.
+            /// SHA Acc is locked on Cold and Warm Resets;
+            /// it is unlocked on Update Reset.
```

Comment:
I don't think this is true. I ran some tests against the RTL, and as best I can tell the update reset does not affect the sha384acc lock state. If the runtime firmware locks the mailbox, it stays locked across the reset. It does appear that if the SoC has the lock, it loses it across the reset.

We don't have any great options to fix #807:

- If we leave this code as is, the runtime firmware can't lock the mailbox before resetting to fix #807, as that would cause the AssumedLockState::NotAcquried attempt here to timeout in the watchdog.

- If we change this code to ShaAccLockState::Acquired for UpdateReset, we are trusting the runtime firmware too much, as an attacker with some control over the runtime firmware could do an update reset without the lock, and have the SoC grab the lock while the ROM assumes it already has the lock. This could be mitigated by doing that verify-random-data-can-be-written-to-register hack I've mentioned before in the assume-acquired case.

Reply:
I updated the comment stating that ROM will expect it to be unlocked on UpdateReset.